### ISSUE TITLE
Filament Sidebar not working correctly with Multi-tenancy + Octane server

### ISSUE THREAD

**Issue by miguilimzero** [laravel-filament/filament#7969]
### Package

filament/filament

### Package Version

v3.0.30

### Laravel Version

v10.19.0

### Livewire Version

v3.0.0-beta.10

### PHP Version

PHP 8.2.9

### Problem description

There are currently two different problems depending on which Octane adapter you're now using:

### Swoole
If you have more than one tenancy and switch between them, the sidebar link will not update accordingly. Example:

- I'm on team ID 1
- Now I switched to team ID 2
- The post resource sidebar link will still point to team ID 1

This happens randomly. Sometimes it updates the links correctly, sometimes not. Most of time the bug will appear.

### Roadrunner
If you keep refreshing the page, the sidebar resources will appear and disappear randomly.

**I could only reproduce at the beginning. I don't know if it was a fault on my side. However, the Swoole issue is easily reproducible.**

### Expected behavior

Everything works seamlessly.

### Steps to reproduce

- Download the example repository
- php artisan migrate
- Set the octane runner you want to test using `OCTANE_SERVER` (Swoole)
- Start the local server with `php artisan octane:start`
- Register your user
- Create team 1 and then create team 2

### Reproduction repository

https://github.com/miguilimzero/filament-tenant-octane-bug-example

**Comment by miguilimzero** [comment]
One thing that I noticed while debugging this issue is that the `mountNavigation()` method from `Panel\Concerns\HasNavigation` is called multiple times using **Swoole**, while with the default web server or Roadrunner, the method is called one time in one request.

- 4 times in one request when the sidebar link is updated correctly.
- 3 times in one request when the sidebar link points to the wrong team ID.

**Comment by binaryfire** [comment]
Just ran into this as well. I'm using Swoole. Here's a video demonstrating the problem: https://www.youtube.com/watch?v=1r7deK__IJI

I've pushed a new version of the example repo to https://github.com/binaryfire/filament-tenant-octane-bug. The packages have been updated and a missing relationship has been fixed. You should be able to just clone it and run `php artisan octane:start` to reproduce. Just make sure you have the Swoole extension installed.

Any chance this could get bumped to high priority? I'm happy to help with testing. Not sure how much help I'll be on the debugging front though.

**Comment by binaryfire** [comment]
Just tested an app that uses Spatie's tenancy package for subdomain multitenancy. The same issue occurs with that too (with Filament's tenancy disabled).

Octane boots once and stays in memory, which means any data cached in memory is persistent and shared across requests. So statics, global variables and singletons can cause problems. This is a good article, in case it helps: https://toluwaniadewale.com/blog/laravel-octane-with-great-performance-comes-great-responsibility

Seems like the sidebar / sidebar item state is being cached? And therefore the first tenant's sidebar that's rendered is re-used by Octane for all subsequent requests.

**Comment by binaryfire** [comment]
@danharrin Spent some time debugging today. `$navigationGroups`, `$navigationItems` and `$isNavigationMounted` in `HasNavigation` are all being preserved between requests. As a quick and dirty test, this fixes the problem:

```
    public function getNavigation(): array
    {
        if ($this->navigationBuilder instanceof Closure) {
            return $this->buildNavigation();
        }

        $this->navigationGroups = [];
        $this->navigationItems = [];
        $this->mountNavigation();
        // if (! $this->isNavigationMounted) {
        //     $this->mountNavigation();
        // }
        
        .....

```

I'm assuming the problem is that Filament calls lots of methods statically. Some key points from this article: https://betterprogramming.pub/the-downsides-of-coding-with-laravel-octane-17f4a7a4ea85:

> If you want to break Laravel Octane, you can even do a simple memory leak by piling up data into a Class static property for each request.
> 
> MyLeaker::$pileUp = Str::random(100);

and:

> **Coding With Laravel Octane in Mind**
> Now that you know how it works, it’s very easy to understand that the application, as a whole, is essentially the same in all aspects across all requests. Changing the state globally in the bootstrapping will rebound on all requests, and outside of it, it will only work for the current one. The application doesn’t die once the request is out, but it will immediately serve the next one that comes into the server.
> 
> With that in mind, here are my recommendations if you plan to create an application with Laravel Octane or create a package that promises the same:
> 
> - It’s safe to register singletons if you expect them to be resolved only once and to stay the same across requests.
> - It’s safe to inject the config if you expect immutability from it.
> - It’s safe to inject the container to resolve a service registered at bootstrap, but not to add or modify something inside it and expect it to persist.
> - Always try to use request() outside controllers.
> - **Never use static properties. Use constants instead when possible.**
> 
> In short, the application state at bootstrapping is different from the application instance at request time. Also, singletons should not be directly tied to requests.
> 


I'm happy to help with getting Octane compatibility happening. But I think it will require some major refactoring?

**Comment by francoism90** [comment]
@binaryfire This one of the reasons I moved away from Octane. I'm not saying it's bad, but you need to be really careful when dealing with static, but even with secure content. It's pretty easy to not get the value you are expecting.

I'm surprised you didn't encounter <https://github.com/livewire/livewire/pull/6324>, because I couldn't even get a simple Livewire component to work in Octane. Maybe this has been improved?

Anyway, I would suggest looking for alternatives, like computed caches, and other things to improve responsiveness.

**Comment by danharrin** [comment]
I understand not using static properties. However, we never use them to store state. Anything that we ever put inside a static property can safely be persisted between requests. So I don't actually think that Filament's static usage is a problem here.

**Comment by binaryfire** [comment]
@francoism90 We've been building Octane apps for 2 years with no problems. As long as you know what to avoid it's fine. We've found that using Octane for local development is the best approach. That way you can pick up on and correct any issues immediately. Apps that run on Octane will always run on FPM, but not always the other way around.

**Comment by binaryfire** [comment]
@danharrin Is there a way to create panels without using a service provider? Similar to how we can add forms and tables to Livewire components. The request object isn't accessible in service providers in Octane so we can't use any domain-related logic in the panel method.

Maybe that'll solve the other issues too.

**Comment by mateusgalasso** [comment]
I've been having problems with Octane and the filament sidebar too. My problem is access, some users can see some menus and others cannot, when a user logs out of the machine and another logs in we have the problem of a user with less access, seeing all the menus that they could not see. Without Octane the problem does not happen

Temporary solution is to use customizable navigation items

```php
 NavigationItem::make('Business Unit')
                    ->url('/admin/business-units')
                    ->icon('heroicon-m-building-storefront')
                    ->visible(fn (): bool => auth()->user()->hasPermissionTo('view_any_business::unit')),
```

**Comment by danharrin** [comment]
Fixed by #9875, I have written notes to detail the solution
